### PR TITLE
Repair texmf bibliography search and update bibliography documentation

### PR DESCRIPTION
**Pull request by danielwe**
* Fixes some bugs that made the texmf bibliography search nonoperative.
* Makes the search recursive (the correct location for a bib file in a texmf tree is typically `$TEXMFHOME/bibtex/bib/bibliography.bib`, not `$TEXMFHOME/bibliography.bib`).
* Brings the bibliography documentation up to date with both the texmf and YAML options.

### CODE REVIEW

File: python3/vim_pandoc/bib/collator.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import os
 import vim
 from glob import glob
+from itertools import chain
```
Comment (alerque) [inline python3/vim_pandoc/bib/collator.py]: Does this extra dependency need to be documented anywhere?

Comment (danielwe) [reply python3/vim_pandoc/bib/collator.py]: I wouldn't think so. `itertools` is part of the python standard library, just like `os`, `glob`, and `subprocess`.

Hpwever, it's easy to rewrite the code to get rid of this dependency.

**Comment by danielwe** [comment]
Just noticed another minor issue in the docs, fixed in the last commit.

**Comment by alerque** [comment]
Nevermind apparrently Github will let me do that on your branch because of this PR. I just squashed the fixup commits for you.

**Comment by alerque** [comment]
...which of course I did _wrong_ and once the mistake was made Github auto-closed this PR and won't let me fix it. Sorry about that. I've merged the branch manually in 53f14ea.

Thanks for the contribution and sorry about the mess I made of this PR!

**Pull request closed without merging**